added Zeus rework

Joecuronium opened the pull request:
Option in the CBA settings to opt-in to the new system
Zeus player will not be glued to the camera and can be tp'd around
ZEUS ACE interact has an option to switch ACRE speaker from unit to curatorCamera and vice-versa

Joecuronium committed:
added Zeus rework

Option in the CBA settings to opt-in to the new system
Zeus player will not be glued to the camera and can be tp'd around
ZEUS ACE interact has an option to switch ACRE speaker from unit to curatorCamera and vice-versa

Bubbus reviewed (commented):
This is good work, thanks Joe.  I have reviewed this by just reading the code so some of my comments may not be reflective of in-game behaviour.

I realise now by looking through this, that I may have misled you by suggesting CBA configuration, when a better option seems to be available.  If we can use the ACE-action configuration to make the new system act like the old system by default, then the "old-mode" wouldn't be needed anymore and we wouldn't need to add a divide between the two.

All that it would need is another ACE toggle, to choose between "teleport zeus player to where the camera is exited" and "leave zeus player alone and let them teleport manually".

If the zeus player isn't glued to the camera anymore then things like AI caching will act different.  I think this is fine because it lets the zeus see which AI is un-cached by the players instead of by the zeus, and they have the AI anchors for other situations.  

If we want to keep that behaviour, I think it would be enough to move an anchor around with the camera instead of the player.  I don't see this as needed though.

Joecuronium committed:
Update customSettings.hpp

Joecuronium committed:
Delete customSettings.sqf

Joecuronium committed:
v2 of zeus rework

-File structure updated
-Zeus UI init moved from scheduled loop to eventhandler
-Added additional actions for self-teleport on cam exit and unit hiding
-Added default variables to mimic legacy behaviour

Joecuronium committed:
changed remoteExec to remoteExecCall

Joecuronium committed:
added check to prevent scripts from running on non-zeus players

Joecuronium committed:
removed redundant code in preparation for rewrite

Joecuronium committed:
Updated closing scripts

-now use CBA functions to wait until the zeus display is really closed and then firing in the unscheduled environment
-experimental detection of RC

Joecuronium commented:
Update to this: Tested it during the Halloween op and it works as expected. I'll clean it up next week and open the PR proper

Joecuronium committed:
cleanup and typo fix

Joecuronium commented:
Cleanup done. Can be reviewed / possibly merged

Joecuronium committed:
more cleanup

Joecuronium committed:
Update fn_exitZeus.sqf

because i am dumb and logic is hard

Joecuronium committed:
Update fn_addZeusActions.sqf

-load order do be weird sometimes

Bubbus commented on components/zeus_ui/fn_addZeusActions.sqf:
```diff
@@ -0,0 +1,151 @@
+
+CLIENT_ONLY;
+
+if(!(player getVariable ["f_var_isZeus",false])) exitWith {}; //Fallback in case the script execution is run on the wrong machine
```
Minor formatting issues - spacing.  This is a total nit-pick comment so if everything else works fine I can clean this up myself before the merge.
```suggestion
if !(player getVariable ["f_var_isZeus", false]) exitWith {}; // Fallback in case the script execution is run on the wrong machine
```

Bubbus commented on components/zeus_ui/fn_addZeusActions.sqf:
```diff
@@ -0,0 +1,151 @@
+
+
+//ACRE actions
+private _script =
```
Descriptive variable names makes it easier to understand what the code does when you read through it later on.
```suggestion
private _speakThroughZeusCamera =
```

Bubbus reviewed (commented):
Looks very good, some minor comments about code style - which is to say that nothing important was found to be wrong.  Let's test this some time and then we can approve and merge it in.

Joecuronium committed:
Update fn_addZeusActions.sqf

added more descriptive variable names

Joecuronium commented:
updated according to review
